### ISSUE TITLE
[BUG]: this may not be a bug but it certainly needs to be fixed

### ISSUE THREAD

**Issue by batman22** [modularml/max#125]
I just installed everything and got it up and running in VS Code. Can run write and run mojo programs not problem. Just can't hit breakpoints?!  I've tried all the settings tweaking and what nots to no avail. Mindblowing this doesn't just work with the mojo extension.   I suspect it is supposed to.  Please help, I really want to try mojo, but if can't do basic debugging, it's a non starter.

   I'm on Mac OSX 14.4, and my VS Code says it's up to date with v. 1.87.2 and I've got mojo 24.1.1 installed.

Help.

Jay

**Comment by mikowals** [comment]
Possible duplicate of modularml/mojo#1924.

**Comment by batman22** [comment]
Yes, that sounds exactly like what I'm experiencing.

**Comment by iamtimdavis** [comment]
@walter-erquinigo @River707 - PTAL when you get a moment

**Comment by River707** [comment]
Thanks for reporting this! We've identified an issue with the currently shipped SDK where we're accidentally stripping symbols used by the debugger for JIT debugging. We're working on a fix, but it will likely only make it into the 24.3 release of Mojo/Max (not the next release, but the one after). Terribly sorry for the issues encountered! 🙇

**Comment by walter-erquinigo** [comment]
Thank for the report and apologies for the broken function. However, I'm going add here some troubleshooting until we get the proper fix out:

Regular non-JIT debugging still working, so if you want to debug the program foo.mojo, you can run the following two commands from a terminal

```
mojo build -O0 -debug-level=full foo.mojo -o /tmp/bin
mojo debug --rpc /tmp/bin 
```

That should start the debug session on VSCode, via the --rpc functionality, without having to JIT. I hope this can get you unblocked.

**Comment by iamtimdavis** [comment]
Please see https://github.com/modularml/mojo/issues/1924 and reopen if you still have issues!
